com.arm.cmsis.pack/src/com/arm/cmsis/pack/rte/examples/RteExampleItem.java

```diff
@@ -237,6 +238,15 @@ public String getLoadPath() {
 		return fLoadPath;	
 	}
 
+	@Override
+	public String getProjectFolder() {
+		if (fProjectFolder == null && isSupported()) {
+			ICpEnvironmentProvider envProvider = CpPlugIn.getEnvironmentProvider();
+			fProjectFolder = envProvider.getProjectFolder(getExample());

```

Comment:
null check is required for envProvider

Reply:
Can envProvider be null? The implementation of getEnvironmentProvider should prevent it from being null and non of other listed methods does the null check.

Reply:
Yes, you are right, the null check is redundant here,  CpPlugIn.getEnvironmentProvider() never returns null.